rfc-parser/src/main/java/tech/pantheon/yanginator/parser/services/GrammarKitRFCUtils.java

```diff
@@ -0,0 +1,342 @@
+                int hexLowerBoundary = Integer.decode(hexLowerBoundaryString);
+                int hexUpperBoundary = Integer.parseInt(originalHexDefWithZero.substring(5).toLowerCase(), 16);
+                int hex = hexLowerBoundary;
+                StringBuilder replacement = new StringBuilder();
+                while (hex <= hexUpperBoundary) {
+                    if (hex == 34 | hex == 92) {
+                        replacement.append(" \"\\").append((char) hex).append("\" |");
+                    } else {
+                        replacement.append(" \"").append((char) hex).append("\" |");
+                    }
+                    hex++;
+                }
+                replacement = new StringBuilder(replacement.substring(0, replacement.length() - 2));
+                line = line.replace(originalHexDef, replacement.toString());
+
+            }
+            result.add(line);
+        }
+
+        return result;
+    }
+
+    /**
+     * Replaces each occurrence of single hexadecimal value with corresponding character.
+     *
+     * @param lines the list of strings in which hexadecimal values will be replaced.
+     * @return the list of strings in which hexadecimal values are replaced by characters.
+     */
+    public static List<String> replaceHexadecimal(List<String> lines) {
+        List<String> result = new ArrayList<>();
+        final Pattern hexPattern = Pattern.compile("(?<hexadecimal>(%x\\d{1,2}\\w))");
+
+        for (String line : lines) {
+            Matcher matcher = hexPattern.matcher(line.trim());
+            while (matcher.find()) {
+                String originalHexDef = matcher.group("hexadecimal").trim();
+                if (isSpecialCharacter(originalHexDef)) {
+                    String replacement = transformSpecialChars(originalHexDef);
+                    line = line.replace(originalHexDef, replacement);
+                } else {
+                    String hex = originalHexDef.substring(2).toLowerCase();
+                    if (hex.length() == 2) {
+                        String replacement = new String(BaseEncoding.base16().lowerCase().decode(hex));
+                        line = line.replace(originalHexDef, "\"" + replacement + "\"");
+                    }
+                }
+            }
+            result.add(line);
+        }
+        return result;
+    }
+
+    /**
+     * Returns {@code true} if the string is hexadecimal representation
+     * of: '\r', '\t', '\n', '"', or '\'.
+     *
+     * @param originalHexDef the string with hexadecimal value
+     * @return {@code true} if given string equals to specified hexadecimal values, {@code false} otherwise
+     */
+    private static boolean isSpecialCharacter(final String originalHexDef) {
+        return originalHexDef.equals("%x0D") | originalHexDef.equals("%x09") | originalHexDef.equals("%x0A") | originalHexDef.equals("%x22") | originalHexDef.equals("%x5C");
```

Comment:
this executes 5 comparisons, always. how about:
switch (originalHexDef) {
  case "%x0D":
  case "%x09":     // and others
     return true;
  default:
     return false;
  }
}

Reply:
or really... see below

Reply:
Fixed.